xdsl/dialects/linalg.py

```diff
@@ -1021,7 +1029,7 @@ def parse(cls, parser: Parser) -> Self:
             input,
             init,
             DenseArrayBase.from_list(i64, permutation),
-            result,
+            result if isa(result, TensorType) else None,
```

Comment:
This still doesn't make sense to me, why input `None` if `result` is a non-tensor type?

Reply:
Then we might silently change the return type when parsing if for whatever reason it's incorrect. Not sure how big a deal that is.

Reply:
Thanks, I've had a chat with Alex about this via Zulip - we have agreed to remove passing the return type to the _init_ method from the parse method and for _init_ to handle it :) I've made the changes here